extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlRolloutHistory.java

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.extended.kubectl.util.deployment.DeploymentHelper;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.AppsV1Api;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1PodTemplateSpec;
+import io.kubernetes.client.openapi.models.V1ReplicaSet;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class KubectlRolloutHistory<ApiType extends KubernetesObject>
+    extends Kubectl.ResourceBuilder<ApiType, KubectlRolloutHistory<ApiType>>
+    implements Kubectl.Executable<ApiType> {
+
+  public static final String CHANGE_CAUSE_ANNOTATION = "kubernetes.io/change-cause";
+
+  public V1PodTemplateSpec getTemplate() {
+    return template;
+  }
+
+  public List<History> getHistories() {
+    return histories;
+  }
+
+  public static class History {
+    private final long revision;
+    private final String changeCause;
+
+    public long getRevision() {
+      return revision;
+    }
+
+    public String getChangeCause() {
+      return changeCause;
+    }
+
+    public History(long revision, String changeCause) {
+      this.revision = revision;
+      this.changeCause = changeCause;
+    }
+
+    @Override
+    public String toString() {
+      return "{revision :" + revision + ", changeCause :" + changeCause + "}";
+    }
+  }
+
+  private long revision;
+
+  private final List<History> histories;
+
+  private V1PodTemplateSpec template;
+
+  @Override
+  public ApiType execute() throws KubectlException {
+    validate();
+    AppsV1Api api = new AppsV1Api(this.apiClient);
```

Comment:
im leaning on always making use of the generic api as best as we can tho, but this will work anyway

```java
GenericKubernetesApi<V1Deployment, V1DeploymentList> deploymentApi = getGenericApi(V1Deployment.class, V1DeploymentList.class);
```

Reply:
Since we need to pass the AppsV1Api instance to the method of DeploymentHelper to get the replicasetlist, I don't think it is necessary to use GenericKubernetesApi.